### PR TITLE
Document the subscription API

### PR DESCRIPTION
**Pull request by ericvw**
This PR adds API documentation for subscriptions.  WebSocket, Socket.IO, and Long-polling documentation is provided.

### CODE REVIEW

**Comment by ericvw** [comment]
Updated PR with latest feedback from 883991e993

**Comment by yupinghu** [comment]
LGTM, another pair of eyes?

File: doc/http-api-guide.md [hunk 1]
```diff
@@ -250,3 +270,94 @@ EOF
...
 ```
+
+
+## Subscription data
+
+Clients may access subscription data via native WebSockets, SocketIO, or
+long-polling over HTTP.
+
+The preferred method to access subscription data is over native WebSockets.
+WebSockets is a web standard for receiving event-driven messages without having
```
Comment (ericlu88) [inline doc/http-api-guide.md:282]: /s receiving duplexing?

File: doc/http-api-guide.md [hunk 2]
```diff
@@ -250,3 +270,94 @@ EOF
...
+that are unable to utilize the WebSocket protocol.
+
+The following sections discuss the details for subscription data over native
+WebSockets, SocketIO, and long-polling over HTTP.
+
+### WebSocket and SocketIO message types
+
+When accesing subscription data via native WebSockets or SocketIO, each
+mechanism shares the same message types.  Note that when using WebSockets, that
+the message comes in the form:
```
Comment (ericlu88) [inline doc/http-api-guide.md:297]: It's a bit confusing here. I prefer to have WebSocket and Socket.IO compare side-by-side so the difference is clear.

File: doc/http-api-guide.md [hunk 3]
```diff
@@ -250,3 +270,94 @@ EOF
...
+
+```javascript
+{
+    'type': /* message type */,
+    'data': /* message data */
+}
+```
+
+### Client to Server
```
Comment (ericlu88) [inline doc/http-api-guide.md]: We should add some notes about which is _type_ and which is _data_ for the following section.

**Comment by ericlu88** [comment]
The websocket part looks good to me, but I think we should also document the long-polling part more(the url route, the payload, and specifically the poll id stuff).

**Comment by ericvw** [comment]
Updated PR addressing @ericlu88's feedback.

**Comment by ericlu88** [comment]
LGTM.

**Pull request merged**